unstructured_ingest/v2/processes/connectors/pinecone.py

```diff
@@ -150,9 +190,21 @@ def run(self, path: Path, file_data: FileData, **kwargs: Any) -> None:
             f" with batch size {self.upload_config.batch_size}"
         )
 
-        pinecone_batch_size = self.upload_config.batch_size
-        for pinecone_batch in batch_generator(elements_dict, pinecone_batch_size):
-            self.upsert_batch(batch=pinecone_batch)
+        max_batch_size = self.upload_config.batch_size
+        batch = []
+        batch_size = 0
+        for element in elements_dict:
+            element_size = len(json.dumps(element))
+            if (len(batch) > 0 and batch_size + element_size > (MAX_PAYLOAD_SIZE - 256)) or len(
```

Comment:
There's already a util to help with this that you can use: [generator_batching_wbytes](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/utils/data_prep.py#L18)

Reply:
nice! I'll use it